src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,211 @@
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get driver by id=" + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getAllDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, c.model, c.manufacturer_id, m.name, m.country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars from driversDB.", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getAllDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateCarStatement
+                         = connection.prepareStatement(query)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in carsDB.", e);
+        }
+        deleteDrivers(car);
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteStatement = connection.prepareStatement(query)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car from DB by id="
+                    + id + ". ", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
```

Comment:
I think it would be better:
````suggestion
    public List<Car> getCarsByDriverId(Long driverId) {
````